### ISSUE TITLE
Object Basics: <Should include information about deep copy vs. shallow copy>

### ISSUE THREAD

**Issue by tman2747** [TheOdinProject/curriculum#28597]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

It would be helpful to introduce the concepts of deep copy vs. shallow copy before starting the 4th assignment in this chapter, particularly the assignment on palindromes. This is because some users might attempt to clean the array and then assign the cleaned array to another variable, subsequently calling reverse() on it.

https://developer.mozilla.org/en-US/docs/Glossary/Shallow_copy
https://developer.mozilla.org/en-US/docs/Glossary/Deep_copy

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-object-basics

### (Optional) Discord Name

tman2747

**Comment by CouchofTomato** [comment]
@TheOdinProject/foundations Any thoughts on this?

**Comment by TheSeagullDev** [comment]
I recently completed this lesson and while doing the exercises stumbled upon resources discussing a deep vs shallow copy, so I definitely agree it would make sense to include something on it.

**Comment by ManonLef** [comment]
Need more team input, but at this point in the curriculum this does not feel right to cover. There could probably be mention of it somewhere later in the curriculum, before to-do but after foundations.

**Comment by mao-sz** [comment]
Closing - see https://github.com/TheOdinProject/curriculum/pull/29554#issuecomment-3288960662
